lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -103,47 +103,21 @@ def identify(self, requirement_or_candidate):
         """
         return requirement_or_candidate.canonical_package_id
 
-    def get_preference(self, *args, **kwargs):
-        # type: (t.Any, t.Any) -> t.Union[float, int]
+    def get_preference(
+        self,
+        identifier: str,
+        resolutions: t.Mapping[str, Candidate],
+        candidates: t.Mapping[str, t.Iterator[Candidate]],
+        information: t.Iterator[t.NamedTuple],
+        backtrack_causes: t.Sequence
+    ) -> t.Union[float, int]:
```

Comment:
Can we use slightly newer syntax here?
```suggestion
    ) -> float | int:
```

Reply:
done